src/converters/labelstograph/graphconverter.py

```diff
@@ -52,4 +56,27 @@ def generate_events(labels: list[Label]) -> list[EventNode]:
         event_labels = [label for label in labels if label.name==event_label_name]
         events.append(EventNode(id=f'E{event_counter}', labels=event_labels))
     
-    return events
\ No newline at end of file
+    return events
+
+def resolve_exceptive_negations(labels: list[Label]) -> list[EventLabel]:
+    """Negated events are handled internally by each node resulting from an event label, but exceptive negations ("Unless A then B") have to be handled manually. This method identifies exceptive negations and identifies all events (in the form of event labels) that are affected by the additional negation.
+
+    parameters:
+        labels -- list of labels generated from the sentence
+    
+    returns: list of labels that are affected by an exceptive negation and hence need to be additionally negated"""
+    exceptive_negations = [label for label in labels if label.name=='Negation' and label.parent==None]
+
+    all_events: list[EventLabel] = [label for label in labels if type(label)==EventLabel]
+    all_events.sort(key=(lambda event: event.begin))
+
+    # determine all additionally negated events, i.e., the event immediately following the exceptive negation plus all additional events that are connected to that event through conjunctions
+    negated_events: list[EventLabel] = []
+    for negation in exceptive_negations:
+        affected_event = [label for label in all_events if label.begin > negation.end][0]
+        negated_events.append(affected_event)
+        while affected_event.successor != None and affected_event.successor.junctor == 'AND':
+            affected_event = affected_event.successor.target
+            negated_events.append(affected_event)
```

Comment:
Could be this handled without nested loops?

```python
events_to_consider = [...]
negated_events = [...]
events_to_handel = [...]

events_to_handle:
   doSomething()
```

Reply:
I will need to look into this.